src/stactools/landsat/utils.py

```diff
@@ -98,3 +98,34 @@ def handle_antimeridian(item: Item, antimeridian_strategy: Strategy) -> None:
         item.geometry = Polygon(merged_coords)
 
     antimeridian.fix_item(item, antimeridian_strategy)
+
+
+def round_coordinates(item: Item, precision: int = 6) -> None:
+    """Rounds an Item's geometry and bbox geographic coordinates.
+
+    Any tuples encountered will be converted to lists.
+
+    Args:
+        item (Item): pystac Item.
+        precision (int): Number of decimal places for rounding. Defaults to 6,
+            which is approximately 1 cm at the equator.
+    """
+
+    def recursive_round(coordinates: List[Any], precision: int) -> List[Any]:
+        for idx, _item in enumerate(coordinates):
+            if isinstance(_item, (list, tuple)):
+                coordinates[idx] = list(_item)  # handle any tuples
+                coordinates[idx] = recursive_round(coordinates[idx], precision)
+            elif isinstance(_item, (int, float)):
+                coordinates[idx] = round(_item, precision)
+            else:
+                raise ValueError(f"Unexpected type: {_item} encountered.")
```

Comment:
```suggestion
            if isinstance(_item, (int, float)):
                coordinates[idx] = round(_item, precision)
            else:
                coordinates[idx] = list(_item)  # handle any tuples
                coordinates[idx] = recursive_round(coordinates[idx], precision)
```

I think we can just round numbers first, and trust we can iterate the rest.

Reply:
Yeah, I like that much better. Thanks.